src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1033,6 +1039,21 @@ public StringBuilder Append(char[] value)
             return this;
         }
 
+        public StringBuilder Append(ReadOnlySpan<char> value)
+        {
+            if (value.Length > 0)
+            {
+                unsafe
+                {
+                    fixed (char* valueChars = &value.DangerousGetPinnableReference())
```

Comment:
Could add an `internal` [`wstrcpy'](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/String.cs#L402-L405) that takes Span and call that instead? /cc @stephentoub